gtfs/spec/en/reference.md

```diff
@@ -32,6 +33,22 @@ This section defines terms that are used throughout this document.
 * **Field optional** - The field column may be omitted from your feed. If you choose to include an optional column, each record in your feed must have a value for that column. You may include an empty string as a value for records that do not have values for the column. Some optional fields permit an empty string as a value. To enter an empty string, just omit any text between the commas for that field. Note that 0 is interpreted as "a string of value 0", and is not an empty string.
 * **Dataset unique** - The field contains a value that maps to a single distinct entity within the column. For example, if a route is assigned the ID **1A**, then no other route may use that route ID. However, you may assign the ID **1A** to a location because locations are a different type of entity than routes.
 
+## Field Types
+
+- **Ad-hoc** - Please see the detail column of the field for the definition of the field type.
+- **Color** - The field contains a color encoded as a six-character hexadecimal number. Refer to https://htmlcolorcodes.com/ to generate a valid value. E.g. "FFFFFF" for white, "000000" for black or "0039A6" for the A,C,E lines in NYMTA.
+- **Enum** - The field contains a value among a set of ad-hoc predefined constants defined in the "Details" column. E.g. the route_type field contains a "0" for tram, a "1" for subway...
+- **ID** - The field contains an identifier which uniquely identifies an entity (therefore  is dataset unique). Its value is not aimed to be displayed to the user, and is a sequence of any UTF-8 characters, but using only printable ASCII characters is recommended.
+- **Positive Integer** - The field contains a non-negative whole number (which includes 0).
```

Comment:
As ugly as it is, I think this field must be named `Non-negative Integer` - "positive" by definition excludes 0.

And, I'd change the definition to `The field contains a the value 0 or a positive whole number.`

Reply:
> "positive" by definition excludes 0.

Haha. Oh God. Not in French Math (aka not in real Math). But yeah, I've heard that's how you do Math. It is so confusing. Zero is obviously positive *and* negative. This is why we have **N*** and **Z***. Anyway... #DontStartMeOnThat #ThisIsWhyWeHave12FieldMedals #LongLiveNAndZStar

Reply:
Anyway. I'll replace "Positiv" by "Non-negative".